OpenSim/Simulation/Model/GeometryPath.h

```diff
@@ -213,36 +201,71 @@ OpenSim_DECLARE_CONCRETE_OBJECT(GeometryPath, ModelComponent);
     void extendInitStateFromProperties(SimTK::State& s) const override;
     void extendAddToSystem(SimTK::MultibodySystem& system) const override;
 
-    // Visual support GeometryPath drawing in SimTK visualizer.
-    void generateDecorations(
-            bool                                        fixed,
-            const ModelDisplayHints&                    hints,
-            const SimTK::State&                         state,
-            SimTK::Array_<SimTK::DecorativeGeometry>&   appendToThis) const
-            override;
-
     void extendFinalizeFromProperties() override;
 
-private:
+protected:
+    virtual void computePath(const SimTK::State& s ) const = 0;
+
+    virtual double calcLengthAfterPathComputation(
+            const SimTK::State&              s,
+            const Array<AbstractPathPoint*>& currentPath) const = 0;
 
-    void computePath(const SimTK::State& s ) const;
-    void computeLengtheningSpeed(const SimTK::State& s) const;
-    void applyWrapObjects(const SimTK::State& s, Array<AbstractPathPoint*>& path ) const;
-    double calcPathLengthChange(const SimTK::State& s, const WrapObject& wo, 
-                                const WrapResult& wr, 
-                                const Array<AbstractPathPoint*>& path) const; 
-    double calcLengthAfterPathComputation
-       (const SimTK::State& s, const Array<AbstractPathPoint*>& currentPath) const;
+    virtual void generateDecorations(
+            bool                                      fixed,
+            const ModelDisplayHints&                  hints,
+            const SimTK::State&                       state,
+            SimTK::Array_<SimTK::DecorativeGeometry>& appendToThis) const override = 0;
 
     void constructProperties();
-    void namePathPoints(int aStartingIndex);
-    void placeNewPathPoint(const SimTK::State& s, SimTK::Vec3& aOffset, 
-                           int index, const PhysicalFrame& frame);
     //--------------------------------------------------------------------------
     // Implement Object interface.
     //--------------------------------------------------------------------------
     /** Override of the default implementation to account for versioning. */
-    void updateFromXMLNode(SimTK::Xml::Element& aNode, int versionNumber = -1) override;
+    void updateFromXMLNode(
+            SimTK::Xml::Element& aNode,
+            int versionNumber = -1) override;
+
+
+
+
+// related to pathpoints, so has to be removed in release
+protected:
```

Comment:
These methods can actually be removed. In the testing I've done these errors where never thrown (and should never be thrown since an old "geometrypath" is explicitly linked to the new "pointbasedPath"